src/storage/src/hummock/sstable/mod.rs

```diff
@@ -158,21 +161,29 @@ impl Sstable {
             true
         };
         if enable_bloom_filter() && self.has_bloom_filter() {
-            let hash = xxh32::xxh32(dist_key, 0);
+            let hash = xxh64::xxh64(dist_key, 0);
             self.may_match_hash(hash)
         } else {
             true
         }
     }
 
     #[inline(always)]
-    pub fn hash_for_bloom_filter(dist_key: &[u8], table_id: u32) -> u32 {
+    pub fn hash_for_bloom_filter_u32(dist_key: &[u8], table_id: u32) -> u32 {
         let dist_key_hash = xxh32::xxh32(dist_key, 0);
+        // congyi adds this because he aims to dedup keys in different tables
         table_id.bitxor(dist_key_hash)
     }
 
     #[inline(always)]
-    pub fn may_match_hash(&self, hash: u32) -> bool {
+    pub fn hash_for_bloom_filter(dist_key: &[u8], table_id: u32) -> u64 {
```

Comment:
If there are some sstable files generated by  previous hummock, it would return an error result because we can not hit the previous bloom-filter with u32-hash. 
It means that we need to calculate different hash-value in query for bloom-filter and xor-filter.

Reply:
> If there are some sstable files generated by previous hummock, it would return an error result because we can not hit the previous bloom-filter with u32-hash.
> It means that we need to calculate different hash-value in query for bloom-filter and xor-filter.

Let's assume that no files are generated by previous hummock 😄

Reply:
ok